src/instance.rs

```diff
@@ -138,6 +138,14 @@ impl<'py> Bound<'py, PyAny> {
     ) -> PyResult<Self> {
         Py::from_owned_ptr_or_err(py, ptr).map(|obj| Self(py, ManuallyDrop::new(obj)))
     }
+
+    #[inline]
+    pub(crate) unsafe fn from_ref_to_ptr<'a>(
```

Comment:
I don't really get this name. How about `from_borrowed_ptr`? 

Same on `BoundRef`

Reply:
Ugh, I couldn't work out a good name either. I didn't feel `borrowed` is quite right here because that's got a very specific meaning in the Python C API to refer to a pointer return value which doesn't carry ownership.

How about just `from_raw`?

Reply:
(I think this API will remain crate-private for the foreseeable future, so the name doesn't have to be perfect 😄)

Reply:
I agree, since this is internal its not super important. I'm also fine with `from_raw`. I think I also get now that you were intending to reference this (unusual) construct `& *mut _`, but yesterday i read it like it was turning a reference _into_ a pointer, while it was doing kind of the opposite 🙈

Reply:
I don't think I have a particular preference among these. If i had to pick i would probably choose between `from_shared_ref_ptr` and `from_ptr_shared_ref`. So I'm fine with your preference. Another option that came to my mind was something like `ref_from_ptr` or `shared_from_ptr`, but I dont't feel strongly about these either.

Reply:
I quite like `ref_from_ptr`, let's go with that 👍